dcmgr/lib/dcmgr/node_modules/maintenance.rb

```diff
@@ -32,6 +34,15 @@ def expire_ip_handles
       }
     end
 
+    def automatic_windows_password_deletion
+      # logger.debug "Checking for windows passwords to delete"
+
+      Dcmgr::Models::Instance.dataset.alives.where('password_will_be_deleted_at IS NOT NULL').expired.each { |handle|
```

Comment:
Remarks here.

1) The lines are a little long. We should try to shorten it if possible. I think 80 colons is a good guide line. A little longer is no problem but it shouldn't be *too* much longer so horizontal scrolling or wrapped lines can be avoided when using split screens in editors.

2) I think it's better not to write SQL code directly if it can be avoided. Though not a problem in this particular case, it's bad practice that could allow for SQL injection if used wrongly. Another reason to avoid it is to stay consistent with the rest of the code base.

3) The variable name `handle` isn't a good name. We're getting an object of `Dcmgr::Models::Instance` for every iteration so we should call it `inst` or `instance`.

I would refactor it like this.

```ruby
expired_insts = Dcmgr::Models::Instance.alives.password_deletion_time_passed
expired_insts.exclude(password_will_be_deleted_at: nil).each { |inst|
  logger.debug "Automatically deleting encrypted password for instance with uuid: %s" %
    inst.canonical_uuid
  inst.delete_windows_password
}
```